### PR TITLE
feature/LHUB039

### PR DESCRIPTION
**Pull request by guilhermedevel**
Full account update endpoint flow implementation

### CODE REVIEW

File: src/main/java/com/godev/linkhubservice/rest/controllers/AccountController.java [hunk 1]
```diff
@@ -1,16 +1,14 @@
 package com.godev.linkhubservice.rest.controllers;
 
-import com.godev.linkhubservice.domain.vo.AccountRequest;
-import com.godev.linkhubservice.domain.vo.AccountResponse;
-import com.godev.linkhubservice.domain.vo.AuthRequest;
-import com.godev.linkhubservice.domain.vo.AuthResponse;
+import com.godev.linkhubservice.domain.vo.*;
```
Comment (jecesario) [inline src/main/java/com/godev/linkhubservice/rest/controllers/AccountController.java]: Recommended not to use wild card

File: src/main/java/com/godev/linkhubservice/rest/controllers/AccountController.java [hunk 2]
```diff
@@ -35,4 +33,7 @@ public interface AccountController {
     })
     @PostMapping(value = "/auth")
     ResponseEntity<AuthResponse> auth(@RequestBody @Valid AuthRequest authRequest);
+
+    @PutMapping
+    ResponseEntity<UpdateAccountResponse> update(@Valid @RequestBody UpdateAccountRequest updateAccountRequest);
```
Comment (jecesario) [inline src/main/java/com/godev/linkhubservice/rest/controllers/AccountController.java]: Arrange annotation order @Valid @RequestBody

File: src/main/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImpl.java [hunk 3]
```diff
@@ -4,10 +4,7 @@
 import com.godev.linkhubservice.domain.exceptions.InvalidJwtException;
 import com.godev.linkhubservice.domain.exceptions.Issue;
 import com.godev.linkhubservice.domain.exceptions.IssueEnum;
-import com.godev.linkhubservice.domain.vo.AccountRequest;
-import com.godev.linkhubservice.domain.vo.AccountResponse;
-import com.godev.linkhubservice.domain.vo.AuthRequest;
-import com.godev.linkhubservice.domain.vo.AuthResponse;
+import com.godev.linkhubservice.domain.vo.*;
```
Comment (jecesario) [inline src/main/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImpl.java]: Recommended not to use wild card

File: src/main/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImpl.java [hunk 4]
```diff
@@ -72,4 +69,16 @@ public ResponseEntity<AuthResponse> auth(AuthRequest authRequest) {
...
+    @Override
+    public ResponseEntity<UpdateAccountResponse> update(UpdateAccountRequest updateAccountRequest) {
+
+        log.info("Starting update of account {}", updateAccountRequest.getName());
+
+        var updateAccountResponse = accountService.update(updateAccountRequest);
+
+        log.info("Account update successful");
+
+        return ResponseEntity.status(HttpStatus.OK).body(updateAccountResponse);
```
Comment (jecesario) [inline src/main/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImpl.java]: Could response just ResponseEntity.ok(updateAccountResponse);

File: src/main/java/com/godev/linkhubservice/services/AccountService.java [hunk 5]
```diff
@@ -1,8 +1,7 @@
 package com.godev.linkhubservice.services;
 
-import com.godev.linkhubservice.domain.vo.AccountRequest;
-import com.godev.linkhubservice.domain.vo.AccountResponse;
-import com.godev.linkhubservice.domain.vo.AuthRequest;
+import com.godev.linkhubservice.domain.models.Account;
+import com.godev.linkhubservice.domain.vo.*;
```
Comment (jecesario) [inline src/main/java/com/godev/linkhubservice/services/AccountService.java]: Recommended not to use wild card

File: src/main/java/com/godev/linkhubservice/services/impl/AccountServiceImpl.java [hunk 6]
```diff
@@ -7,12 +7,11 @@
 import com.godev.linkhubservice.domain.exceptions.RuleViolationException;
 import com.godev.linkhubservice.domain.models.Account;
 import com.godev.linkhubservice.domain.repository.AccountRepository;
-import com.godev.linkhubservice.domain.vo.AccountRequest;
-import com.godev.linkhubservice.domain.vo.AccountResponse;
-import com.godev.linkhubservice.domain.vo.AuthRequest;
+import com.godev.linkhubservice.domain.vo.*;
```
Comment (jecesario) [inline src/main/java/com/godev/linkhubservice/services/impl/AccountServiceImpl.java]: Recommended not to use wild card

File: src/main/java/com/godev/linkhubservice/services/impl/AccountServiceImpl.java [hunk 7]
```diff
@@ -88,20 +87,57 @@ public UserDetails auth(AuthRequest authRequest) {
...
-        log.info("E-mail {} found", email);
+    }
 
-        return AccountResponse
+    @Override
+    public UpdateAccountResponse update(UpdateAccountRequest updateAccountRequest) {
+        var userDetails = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        var account = findByEmail(userDetails.getUsername());
+
+        log.info("Verifying if e-mail {} already registered", updateAccountRequest.getEmail());
```
Comment (jecesario) [inline src/main/java/com/godev/linkhubservice/services/impl/AccountServiceImpl.java]: This log make sense?

File: src/test/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImplTest.java [hunk 8]
```diff
@@ -4,34 +4,28 @@
 import com.godev.linkhubservice.domain.exceptions.BadRequestException;
 import com.godev.linkhubservice.domain.exceptions.Issue;
 import com.godev.linkhubservice.domain.exceptions.IssueEnum;
-import com.godev.linkhubservice.services.AccountService;
-import com.godev.linkhubservice.helpers.AccountRequestMockBuilder;
-import com.godev.linkhubservice.helpers.AccountResponseMockBuilder;
-import com.godev.linkhubservice.helpers.AuthRequestMockBuilder;
-import com.godev.linkhubservice.helpers.AuthResponseMockBuilder;
+import com.godev.linkhubservice.helpers.*;
```
Comment (jecesario) [inline src/test/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImplTest.java]: Recommended not to use wild card

File: src/test/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImplTest.java [hunk 9]
```diff
@@ -4,34 +4,28 @@
...
 import static com.godev.linkhubservice.domain.constants.IssueDetails.GENERATE_AUTH_TOKEN_ERROR;
 import static com.godev.linkhubservice.domain.constants.IssueDetails.INVALID_CREDENTIALS_ERROR;
-import static com.godev.linkhubservice.domain.constants.ValidationConstants.EMAIL_FORMAT_ERROR;
-import static com.godev.linkhubservice.domain.constants.ValidationConstants.EMAIL_LENGTH_ERROR;
-import static com.godev.linkhubservice.domain.constants.ValidationConstants.EMAIL_REQUIRED_ERROR;
-import static com.godev.linkhubservice.domain.constants.ValidationConstants.NAME_LENGTH_ERROR;
-import static com.godev.linkhubservice.domain.constants.ValidationConstants.NAME_REQUIRED_ERROR;
-import static com.godev.linkhubservice.domain.constants.ValidationConstants.PASSWORD_FORMAT_ERROR;
-import static com.godev.linkhubservice.domain.constants.ValidationConstants.PASSWORD_REQUIRED_ERROR;
+import static com.godev.linkhubservice.domain.constants.ValidationConstants.*;
```
Comment (jecesario) [inline src/test/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImplTest.java]: Recommended not to use wild card

File: src/test/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImplTest.java [hunk 10]
```diff
@@ -332,4 +326,182 @@ void authRequestPasswordFieldHasInvalidLength() throws Exception {
                 .andExpect(content().json(objectMapper.writeValueAsString(new Issue(IssueEnum.ARGUMENT_NOT_VALID, PASSWORD_FORMAT_ERROR))));
 
     }
+
+    @BeforeEach
+    void setup(){
```
Comment (jecesario) [inline src/test/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImplTest.java]: setup method maybe could come before all methods

File: src/test/java/com/godev/linkhubservice/services/impl/AccountServiceImplTest.java [hunk 11]
```diff
@@ -4,32 +4,36 @@
...
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
 import java.util.List;
 import java.util.Optional;
 
-import static com.godev.linkhubservice.domain.constants.IssueDetails.EMAIL_EXISTS_ERROR;
-import static com.godev.linkhubservice.domain.constants.IssueDetails.EMAIL_NOT_FOUND_ERROR;
-import static com.godev.linkhubservice.domain.constants.IssueDetails.INVALID_CREDENTIALS_ERROR;
+import static com.godev.linkhubservice.domain.constants.IssueDetails.*;
```
Comment (jecesario) [inline src/test/java/com/godev/linkhubservice/services/impl/AccountServiceImplTest.java]: wild card

File: src/test/java/com/godev/linkhubservice/services/impl/AccountServiceImplTest.java [hunk 12]
```diff
@@ -4,32 +4,36 @@
...
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.Mockito.times;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
```
Comment (jecesario) [inline src/test/java/com/godev/linkhubservice/services/impl/AccountServiceImplTest.java]: wild card

**Review by jecesario — CHANGES_REQUESTED** [review]

File: src/main/java/com/godev/linkhubservice/rest/controllers/AccountController.java (hunk 1 above)
Comment (guilhermedevel) [reply src/main/java/com/godev/linkhubservice/rest/controllers/AccountController.java]: check

File: src/main/java/com/godev/linkhubservice/rest/controllers/AccountController.java (hunk 2 above)
Comment (guilhermedevel) [reply src/main/java/com/godev/linkhubservice/rest/controllers/AccountController.java]: Check! Did you see that in the register method the order of these annotations is also changed? Should I exchange them too?

File: src/main/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImpl.java (hunk 3 above)
Comment (guilhermedevel) [reply src/main/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImpl.java]: Checked

File: src/main/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImpl.java (hunk 4 above)
Comment (guilhermedevel) [reply src/main/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImpl.java]: Checked

File: src/main/java/com/godev/linkhubservice/services/AccountService.java (hunk 5 above)
Comment (guilhermedevel) [reply src/main/java/com/godev/linkhubservice/services/AccountService.java]: checked

File: src/main/java/com/godev/linkhubservice/services/impl/AccountServiceImpl.java (hunk 6 above)
Comment (guilhermedevel) [reply src/main/java/com/godev/linkhubservice/services/impl/AccountServiceImpl.java]: checked

File: src/main/java/com/godev/linkhubservice/services/impl/AccountServiceImpl.java (hunk 7 above)
Comment (guilhermedevel) [reply src/main/java/com/godev/linkhubservice/services/impl/AccountServiceImpl.java]: I have changed to:
log.info("Validating request fields");

File: src/test/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImplTest.java (hunk 8 above)
Comment (guilhermedevel) [reply src/test/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImplTest.java]: checked

File: src/test/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImplTest.java (hunk 9 above)
Comment (guilhermedevel) [reply src/test/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImplTest.java]: checked

File: src/test/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImplTest.java (hunk 10 above)
Comment (guilhermedevel) [reply src/test/java/com/godev/linkhubservice/rest/controllers/impl/AccountControllerImplTest.java]: checked

File: src/test/java/com/godev/linkhubservice/services/impl/AccountServiceImplTest.java (hunk 11 above)
Comment (guilhermedevel) [reply src/test/java/com/godev/linkhubservice/services/impl/AccountServiceImplTest.java]: checked

File: src/test/java/com/godev/linkhubservice/services/impl/AccountServiceImplTest.java (hunk 12 above)
Comment (guilhermedevel) [reply src/test/java/com/godev/linkhubservice/services/impl/AccountServiceImplTest.java]: checked

**Review by jecesario — APPROVED** [review]

**Pull request merged**